### PR TITLE
FEAT: Support replica

### PR DESCRIPTION
**Pull request by codingl2k1**

### CODE REVIEW

File: xinference/core/utils.py [hunk 1]
```diff
@@ -51,3 +51,12 @@ def wrapped(*args, **kwargs):
         return wrapped
 
     return decorator
+
+
+def iter_replica_model_uid(model_uid, replica):
```
[the remark this answers is not in the document]
Comment (codingl2k1) [reply xinference/core/utils.py]: Fixed. Thanks.

File: xinference/core/supervisor.py [hunk 2]
```diff
@@ -197,29 +220,59 @@ async def _check_dead_nodes(self):
...
+        worker_ref = self._replica_model_uid_to_worker[replica_model_uid]
+        return await worker_ref.get_model(model_uid=replica_model_uid)
 
     @log_async(logger=logger)
     async def describe_model(self, model_uid: str) -> Dict[str, Any]:
-        if model_uid not in self._model_uid_to_worker:
+        if model_uid not in self._model_uid_to_replica:
             raise ValueError(f"Model not found in the model list, uid: {model_uid}")
+        replica = self._model_uid_to_replica[model_uid]
+        replica_model_uid = random.choice(
```
[the remark this answers is not in the document]
Comment (codingl2k1) [reply xinference/core/supervisor.py:266]: `ModelGroup` requires a lot of code modification. I think we can introduce a `ModelGroup` if current implementation can't meet the requirements.

**Comment by UranusSeven** [comment]
The 'model' attribute in responses should not be the UID of replica models.

How about adding two extra attributes, 'total_replicas' and 'replica_index', to the base class of LLMs?

By doing so, the current model UID remains as it is, and we can use 'total_replicas' and 'replica_index' to differentiate the replicas.

https://github.com/xorbitsai/inference/blob/5562569f4a63e2dec020fae78c83a8dd1185e9d7/xinference/model/llm/core.py

**Comment by UranusSeven** [comment]
How about also adding an option to xinference/deploy/cmdline.py::model_launch?

File: xinference/tests/test_client.py [hunk 3]
```diff
@@ -75,6 +75,28 @@ def test_client_for_embedding(setup):
...
 
+def test_replica_model(setup):
+    endpoint, _ = setup
+    client = Client(endpoint)
+    assert len(client.list_models()) == 0
+
+    model_uid = client.launch_model(
+        model_name="orca", model_size_in_billions=3, quantization="q4_0", replica=2
+    )
+    assert len(client.list_models()) == 2
```
Comment (UranusSeven) [inline xinference/tests/test_client.py:86]: I think users are not supposed to see the replica model uid. Thus, list_models should return only one item.

File: xinference/core/utils.py [hunk 4]
```diff
@@ -51,3 +53,25 @@ def wrapped(*args, **kwargs):
         return wrapped
 
     return decorator
+
+
+def iter_replica_model_uid(model_uid: str, replica: int) -> Generator[str, None, None]:
```
[the remark this answers is not in the document]
Comment (codingl2k1) [reply xinference/core/utils.py:58]: Test added, thanks.

File: xinference/tests/test_client.py (hunk 3 above)
Comment (codingl2k1) [reply xinference/tests/test_client.py]: Fixed, thanks.

**Comment by codingl2k1** [comment]
> How about also adding an option to xinference/deploy/cmdline.py::model_launch?

I added an option `--replica`, `-r` to the cmdline.

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**